src/libs/migrations/PersonalDetailsByAccountID.js

```diff
@@ -59,155 +75,165 @@ function getDeprecatedPersonalDetailsFromOnyx() {
-                    Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because the reportAction was empty`);
-                    return;
-                }
-
-                const newReportAction = reportAction;
-
-                if (lodashHas(reportAction, ['originalMessage', 'oldLogin']) && !lodashHas(reportAction, ['originalMessage', 'oldAccountID'])) {
-                    reportActionsWereModified = true;
-                    const oldAccountID = _.get(oldPersonalDetails, [reportAction.originalMessage.oldLogin, 'accountID']);
-                    if (oldAccountID) {
-                        newReportAction.originalMessage.oldAccountID = oldAccountID;
-                    } else {
-                        Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because originalMessage.oldAccountID not found`);
-                        return;
-                    }
-                }
-
-                if (lodashHas(reportAction, ['originalMessage', 'newLogin']) && !lodashHas(reportAction, ['originalMessage', 'newAccountID'])) {
-                    reportActionsWereModified = true;
-                    const newAccountID = _.get(oldPersonalDetails, [reportAction.originalMessage.newLogin, 'accountID']);
-                    if (newAccountID) {
-                        newReportAction.originalMessage.newAccountID = newAccountID;
-                    } else {
-                        Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because originalMessage.newAccountID not found`);
-                        return;
-                    }
-                }
-
-                if (lodashHas(reportAction, ['accountEmail']) && !lodashHas(reportAction, ['accountID'])) {
-                    reportActionsWereModified = true;
-                    const accountID = _.get(oldPersonalDetails, [reportAction.accountEmail, 'accountID']);
-                    if (accountID) {
-                        newReportAction.accountID = accountID;
-                    } else {
-                        Log.info(`[Migrate Onyx] PersonalDetailsByAccountID migration: removing reportAction ${reportActionID} because accountID not found`);
+            // We migrate reportActions to have the new accountID-based data if they don't already.
+            // If we are not able to get the accountID from personalDetails for some reason, we will just clear the reportAction
+            // and let it be fetched from the API next time they open the report and scroll to that action.
+            // We do this because we know the reportAction from the API will include the needed accountID data.
+            if (!oldReportActions) {
+                Log.info('[Migrate Onyx] Skipped PersonalDetailsByAccountID reportActions migration because there were no reportActions');
+            } else {
```

Comment:
NAB but can we flatten this? No need for the else condition. Or just remove the log that we are skipping the migration (we don't have a similar log for oldPolicyMemberList being empty)

(this is just to clear the diff, it's hard to read but I assume nothing changed in the below code)

Reply:
Hmm, yeah that log doesn't really do much for us. I'm going to remove it.

Reply:
Ok I updated it. Will be easier to review if you choose the "remove whitespace" option.

Reply:
Thank you!